### ISSUE TITLE
Saltpad setup help on OpenBSD - no pam - ldap AD auth is working

### ISSUE THREAD

**Issue by workingmanrob** [Lothiraldan/saltpad#185]
I have a salt master setup running well on a OpenBSD 6.0 amd64 VM and wanted to add saltpad web interface but have hit a few snags.

First and foremost I think is the lack of pam on OpenBSD systems but I already had a samba4 instance ready to be an AD DC so I got that up and running and tried to use ldap eauth instead.  The ldap auth does seem to work as expected using the curl test commands from install docs.
```
# curl -i http://localhost:8000/
HTTP/1.1 200 OK
Content-Length: 99
Server: TornadoServer/4.3
Etag: "aeff1152c7d469b8706b951c530e056620ee4624"
Date: Mon, 05 Dec 2016 02:42:54 GMT
Access-Control-Allow-Origin: *
Content-Type: application/json

{"clients": ["runner", "runner_async", "local_async", "local", "local_batch"], "return": "Welcome"}
```
Then with auth test..
```
# curl -i -H accept=application/json -d username=administrator -d password=LETMEIN -d eauth=ldap http://localhost:8000/login 
HTTP/1.1 200 OK
Date: Mon, 05 Dec 2016 02:46:51 GMT
Access-Control-Allow-Origin: *
Content-Type: application/json
Content-Length: 228
Server: TornadoServer/4.3

{"return": [{"perms": [".*", "@runner", "@wheel"], "start": 1480906011.618547, "token": "82f9f65ba156d73dc7c785f588186241f7076199d94a85445cac3dbdbcca674e", "expire": 1480949211.618612, "user": "administrator", "eauth": "ldap"}]}
```
So I proceed to setup the nginx web server (haven't done SSL yet) and I see the login page but it does not accept administrator login.

The login page just says "Invalid credentials" when using the same as the above curl test.

Is there some other switch I need to make in settings.json or elsewhere to enable ldap saltpad auth?

TIA!

**Comment by workingmanrob** [comment]
```
# salt --versions-report
Salt Version:
           Salt: 2016.3.1
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.3
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.24.0
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.12 (default, Jul 25 2016, 16:14:54)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.3
            ZMQ: 4.1.4
 
System Versions:
           dist:   
        machine: amd64
        release: 6.0
         system: OpenBSD
        version: Not Installed
```

**Comment by workingmanrob** [comment]
Solved my own issue.

Within static/app.js there is a hardcoded EAUTH="pam" value which after I changed to EAUTH="ldap" login works like a charm.

Not a java scripter so I'm not sure how to handle both cases but this minor mod got things working for me.

Thanks very much for the webUI... Off to play with my new saltpad :smiley:

**Comment by workingmanrob** [comment]
To bad I can't seem the solved label button...

**Comment by workingmanrob** [comment]
In case someone else tracks this one down the proper way to switch to LDAP auth is now done in the settings.json file.  I did not find this documented anywhere per se or in install instructions.  I just happened to spot the bullet entry in the pre-alpha3 notes.

- Make the external_auth configurable in the settings. It is optional and the default value is 'pam'.

Still had no idea how this should be set until I found settings.json.sample had this example:

    "EAUTH": "pam", # The configured external_auth system

So instead add this line to settings.json and you will be using ldap auth.

    "EAUTH": "ldap",
